### DISCUSSION TITLE
Ability to filter by fellowship

### DISCUSSION THREAD

**Discussion by recovery20** [code4recovery/12-step-meeting-list#960] [Ideas]
I saw that we can create custom meeting types, but are we able to add additional columns for fellowships so that we can allow users to filter by AA, NA, Al-Anon, etc.?

**Comment by tim-rohrer** [comment]
I can't really think of any way to do that other creating a custom type for each one. That should let you have some filtering capability.

I guess I'm a little surprised that we've not been asked this before, but it is the first time I've seen the question :-)

Are you feeding (or planning to feed) AA meetings into Meeting Guide. That may pose a problem if you're using a single website for multiple programs.

**Reply by recovery20** [reply]
What I initially wanted to do was create a clean, intuitive experience to find meetings across multiple fellowships. I didn’t plan on uploading to meeting guide. But what I wanted to do might be a little challenging because the goal was to get city/fellowship meeting lists to rank as individual urls I.e. /meetings/aa/fl/south-palm-beach/boca-raton/

This way each level feeds back into the parent, like breadcrumbs. And when someone searches for [meeting type] in [city], the corresponding results page would rank. 

I reached out to another dev (because I’m not one lol) and he said doing this would be difficult because the plug-in is JS powered and not located in the DOM. I was thinking about server side rewrites, but that sounds complicated. 

For instance, I wanted this https://recoverymeetingsnearme.com/meetings/?tsml-day=1 to display https://recoverymeetingsnearme.com/meetings/monday or something along those lines.

**Reply by tim-rohrer** [reply]
I'm coming up with a couple of potential solutions, but I don't believe any of them are ideal, nor will they do everything you want.

TSML is written in PHP (server) and JS (client), and I do think the most flexible solution would involve adding at least one more database field (to track the recovery program) in the meeting data, and then adjust front end code from there. But yeah, that would be quite a bit of coding.

This might be easier to do with forks of our Google Sheets and TSML-UI. But again, you'd need someone with JS/TS/React skills.

Another TSML option might be to treat each recovery program as a district or region (perhaps in combination with custom types). I have not done anything like this personally so I'm not sure how well it would work.

**Comment by kiyote33** [comment]
FYI: I use the following code in the functions.php file of a child theme for a couple of fellowship/alano type websites I support: 
`if (function_exists('tsml_custom_types')) {
	tsml_custom_types(array(
		'AA' => 'Alcoholics Anonymous',
		'CA' => 'Cocaine Anonymous',
		'CMA' => 'Crystal Meth Anonymous',	
		'CODA' => 'Co-Dependents Anonymous',
		'GA' => 'Gamblers Anonymous',
		'MA' => 'Marijuana Anonymous',
		'NA' => 'Narcotics Anonymous',
		'OA' => 'Overeaters Anonymous',
		'OCOA' => 'Adult Children of Alcoholics',
		'SAA' => 'Sex Addicts Anonymous',
		'TA' => 'Trauma Anonymous',
		'AL-ANON' => 'AL-ANON Family Groups', 
		'NAR-ANON' => 'Nar-Anon Family Groups', 
	));
	tsml_types(array(
		'O' => 'Open meetings are available to anyone interested in this program of recovery. Non-program visitors may attend an open meeting as an observer.',
		'C' => 'Closed meetings are restricted to those who have a problem related to their alcoholism/addiction and �have a desire to stop.�',
	));
}
`
It allows for filtering by program in the types field.
